backend/controllers/analytics.js

```diff
@@ -65,3 +69,50 @@ module.exports.application = (req, res) => {
             });
         });
 };
+
+module.exports.focusItem = async (req, res) => {
+    var students = [];
+    if(req.user.teacher_id == undefined){
+        return res.status(401).json({
+            status: 'error',
+            message: "You don't have permission for this"
+        });
+    }
+    var analytics = {};
+    if(req.body.student != undefined || req.body.students != undefined){
```

Comment:
Rather than compare to `undefined` directly, let's use JavaScript's type system to our advantage since that will give us a `null` check as well.

```suggestion
    if (!req.body.student || !req.body.students) {
```

Reply:
Wouldn't this `if` statement always get executed unless both `student` and `students` are provided in the body? Shouldn't we have an `and` statement in this if statement?

Reply:
I think the equivalent to what I wrote would be `if(req.body.student || req.body.students)`, which should execute as long as one is defined. I will change it to that.